c3d/src/C3d.cpp

```diff
@@ -432,16 +433,127 @@ class C3dParameterAttrInfo : public C3dSimpleSpelling, C3dDiagOnStruct {
 
         State = SeenExpr;
         return AttributeApplied;
+     }
+
+     default:
```

Comment:
the C++ compiler should be able to prove exhaustivity of the case here... did you get a warning with your compiler?

Reply:
I did actually, but now I see it was GCC:
```
/home/guido/r/everparse/c3d/src/C3d.cpp: In member function ‘virtual clang::ParsedAttrInfo::AttrHandling {anonymous}::C3dParameterAttrInfo::handleDeclAttribute(clang::Sema&, clang::Decl*, const clang::ParsedAttr&) const’:
/home/guido/r/everparse/c3d/src/C3d.cpp:442:3: warning: control reaches end of non-void function  -Wreturn-type]
  442 |   }
      |   ^
```
I suppose it's more defensive against badly-casted enum values?

Reply:
Neat, thanks!

Reply:
ah I'm compiling the plugin with clang, which is happy here -- any thoughts on whether we should require clang?

Reply:
Let's call clang the canonical compiler? I'll switch to it here, but could also check now and then for GCC.